qa/integration-tests/src/test/java/io/camunda/zeebe/it/multitenancy/MultiTenancyOverIdentityIT.java

```diff
@@ -0,0 +1,581 @@
+              .newDeployResourceCommand()
+              .addProcessModel(process, "process.bpmn")
+              .tenantId("tenant-b")
+              .send();
+
+      // then
+      assertThat(result)
+          .failsWithin(Duration.ofSeconds(10))
+          .withThrowableThat()
+          .withMessageContaining("PERMISSION_DENIED")
+          .withMessageContaining(
+              "Expected to handle gRPC request DeployResource with tenant identifier `tenant-b`")
+          .withMessageContaining("but tenant is not authorized to perform this request");
+    }
+  }
+
+  @Test
+  void shouldAuthorizeCreateProcessInstance() {
+    // given
+    final long processDefinitionKey;
+    try (final var client = createZeebeClient(ZEEBE_CLIENT_ID_TENANT_A_AND_B)) {
+      processDefinitionKey =
+          client
+              .newDeployResourceCommand()
+              .addProcessModel(process, "process.bpmn")
+              .tenantId("tenant-a")
+              .send()
+              .join()
+              .getProcesses()
+              .stream()
+              .map(Process::getProcessDefinitionKey)
+              .findFirst()
+              .orElseThrow();
+
+      // when
+      final Future<ProcessInstanceEvent> result =
+          client
+              .newCreateInstanceCommand()
+              .processDefinitionKey(processDefinitionKey)
+              .tenantId("tenant-a")
+              .send();
+
+      // then
+      assertThat(result)
+          .describedAs(
+              "Expect that process instance can be created as the client has access process of tenant-a")
+          .succeedsWithin(Duration.ofSeconds(10));
+    }
+  }
+
+  @Test
+  void shouldNotFindOtherTenantsProcess() {
+    // given
+    final long processDefinitionKey;
+    try (final var client = createZeebeClient(ZEEBE_CLIENT_ID_TENANT_A)) {
+      processDefinitionKey =
+          client
+              .newDeployResourceCommand()
+              .addProcessModel(process, "process.bpmn")
+              .tenantId("tenant-a")
+              .send()
+              .join()
+              .getProcesses()
+              .stream()
+              .map(Process::getProcessDefinitionKey)
+              .findFirst()
+              .orElseThrow();
+    }
+
+    try (final var client = createZeebeClient(ZEEBE_CLIENT_ID_TENANT_B)) {
+      // when
+      final Future<ProcessInstanceEvent> result =
+          client
+              .newCreateInstanceCommand()
+              .processDefinitionKey(processDefinitionKey)
+              .tenantId("tenant-b")
+              .send();
+
+      // then
+      assertThat(result)
+          .failsWithin(Duration.ofSeconds(10))
+          .withThrowableThat()
+          .withMessageContaining("NOT_FOUND")
+          .withMessageContaining("Expected to find process definition with key")
+          .describedAs("Process definition should exist for tenant-a but not for tenant-b")
+          .withMessageContaining("and tenant ID 'tenant-b', but none found");
```

Comment:
:x: We discussed that we would remove the `and tenant ID 'tenant-b'` from the "look-up" error messages. It would be better if we remove this now before it starts failing. (see [discussion](https://camunda.slack.com/archives/C05Q0EXE9V4/p1695093631938229)).